control-services/cli/vater.py

```diff
@@ -2,41 +2,55 @@
 from config import Config
 from jenkins import Jenkins
 from gitea import Gitea
-from semaphore import Semaphore
+from semaphore import Semaphore, SemaphoreTaskArgumentError
 from vDocker import VDocker
 import getpass as gp
 
 def main():
     p = Parser()
     c = Config(p.args.configPath, p.args.envPath)
     p.completeParser(c.cfg["service_list"])
-
-    if p.args.command == 'init':
-        init(c, p.args)
-    elif p.args.command == 'task':
-        task(c, p.args)
-    elif p.args.command == 'sync':
-        sync(c, p.args)
-    elif p.args.command == 'config':
-        config(c, p.args)
-    elif p.args.command == 'stop':
-        stop(c, p.args)
-    elif p.args.command == 'restart':
-        restart(c, p.args)
-    elif p.args.command == 'clean':
-        clean(c, p.args)
-    elif p.args.command == 'access':
-        access(c, p.args)
+    try:
+        if p.args.command == 'init':
+            init(c, p.args)
+        elif p.args.command == 'task':
+            task(c, p.args)
+        elif p.args.command == 'sync':
+            sync(c, p.args)
+        elif p.args.command == 'config':
+            config(c, p.args)
+        elif p.args.command == 'stop':
+            stop(c, p.args)
+        elif p.args.command == 'restart':
+            restart(c, p.args)
+        elif p.args.command == 'clean':
+            clean(c, p.args)
+        elif p.args.command == 'access':
+            access(c, p.args)
+    except CommandArgumentsError as e:
+        p.error(e)
+    
 
 def init(config, args):
     return
 
 def task(config, args):
+    # Use optional arguments so task status checking can be implemented as a future feature enhancement
+    if args.templateAlias is not None:
+        runSemaphoreTask(config, args)
+
+def runSemaphoreTask(config, args): 
     s = Semaphore(config)
-    loginSemaphore(s)
-    s.runTask(
-        args.name, args.classID, args.size
-    )
+    if args.projectName is None:
+        raise CommandArgumentsError('Use of the --template option requires specifying a project. See "vater task -h"') 
```

Comment:
I'm not sure I agree with this. I was a fan of the error handling implemented in `semaphore.py`, but this error feels like it should be handled by using `argparse` [arguments](https://docs.python.org/3/library/argparse.html#the-add-argument-method) like `choices` and `required`. 

```
choices - A container of the allowable values for the argument.
required - Whether or not the command-line option may be omitted (optionals only).
```

I also think there's some discussion to be done about maybe throwing / handling exceptions directly in `completeParser.py`. I am hesitant about handling exceptions of this type directly in `vater.py` because at first glance I feel like it might bloat up to code that is primarily error handling, and hides some of the readability for `vater.py`'s purpose, especially once error handling is appropriately implemented. 

My gut reaction is to split the error handling into a different PR to discuss our long term approach / strategy, but I'm not super opposed to anything here. Error handling code is clear / very much needed.

Reply:
Upon reviewing this section of the code I agree this could have been solved more elegantly. Part of what lead me to this result was that after `vater task`, I wanted to be able to either start a new task or (as a future feature to implement) look up the status of a running task, so we don't know if the argument after task should be: 
1. a project name and a template in that project
2. a keyword like `status` and an ID

However, I think a better solution than what I have currently implemented here would be to introduce a new keyword position, like `vater task <command> [...<other_command_parameters]`, where `command` would be something like `run` if the goal is to run a task, `status` to check the status of a task, etc.

For example, `vater task run Management "Create Class" [...etc.]` or  `vater task status <task_id>`

I'll implement these changes in this PR. Thank you for the suggestion.